### DISCUSSION TITLE
Using inputs when specifying Dockerfile name in the custom Docker action

### DISCUSSION THREAD

**Discussion by dingo-d** [community/community#26939] [Actions]
<p>Is it possible to use inputs when specifying the image name in the custom Docker action?</p>
<p>Something like:</p>
<pre><code>runs:
  using: 'docker'
  image: 'Dockerfile-${{ inputs.php}}'
  args:
    - ${{ inputs.php}}
</code></pre>
<p>I’ve read that it’s not possible to use them in the uses part:</p>
<p><a href="https://github.community/t/expressions-in-docker-uri/16271/5">https://github.community/t5/GitHub-Actions/Expressions-in-Docker-URI/m-p/34067#M1778</a></p>
<p>But this is using them in the image name part. That way I could have multiple Dockerfiles in my repo, for instance for different PHP and Node.js versions, so that I don’t need to create N number of repos for my custom actions.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/dingo-d">@dingo-d</a> ,</p>
<p>Currently it’s not supported to use inputs to specify docker image name.</p>
<p>As an alternative, in your action repository, you can create some subfolders to store your different actions. structure as below:</p>
<pre><code>repo
  - action-a
      - Dockerfile
      - action.yaml
      - entrypoint.sh
  - action-b
      - Dockerfile
      - action.yaml
      - entrypoint.sh
</code></pre>
<p>And specify folder name of your action in workflow yaml file.</p>
<p><img src="https://user-images.githubusercontent.com/52772761/181100886-5c85dd7b-7a99-4197-9d3e-b2e25968f608.png" alt="actions.png" data-base62-sha1="gOho64g1STdvHrcwI6d7yUSS8fG" width="552" height="201"></p>
<p>Thanks.</p>

**Comment by dingo-d** [comment]
<p>EDIT: Oh in my custom actions I can create multiple directories to store different dockers and then when I call my action those will be executed automatically! Will try that, thanks!</p>

**Comment by dingo-d** [comment]
<p>How do I specify the folder of the action in my workflow file?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/dingo-d">@dingo-d</a> ,</p>
<p>In the workflow file, you need to add folder after your repo, for my action:</p>
<pre><code>- name: use action
        uses: weide-zhou/hello-github-actions/&lt;font color="#FF0000"&gt;action-a&lt;/font&gt;@master
        with:
          who-to-greet: wadeaction1
      - uses: weide-zhou/hello-github-actions/&lt;font color="#FF0000"&gt;action-b&lt;/font&gt;@master
        with:
          who-to-greet: wadeaction2
</code></pre>
<p>My workflow here: <a href="https://github.com/weide-zhou/ticket13/runs/695896509?check_suite_focus=true" rel="nofollow noopener">https://github.com/weide-zhou/ticket13/runs/695896509?check_suite_focus=true</a></p>
<p>Thanks.</p>

**Comment by dingo-d** [comment]
<p>That di the trick! Thanks!</p>
